added a pipeline that extracts exemption pages urls for all exemptions publishers

OriHoch opened the pull request:
### changelog
* added `./procurement/tenders/exemptions/publisher-urls` pipeline
  * `add_publisher_urls_resource` - a processor that gets the publishers resource and extracts all the exemption urls for each publisher id
  * `ExemptionsPublisherScraper` - a class that handles the low-level of scraping and iterating over the exemptions search result pages
* removed the dependency on the datapackage-pipelines branch with support for streaming text resources (it is now on master)
* added `ResourceFilterProcessor` class which allows code reuse for common use-cases of having a processor that filters / modifies resources - used by both the publisers and publisher-urls pipelines
* added __init__ files to allow having classes in the pipeline directories

akariv commented on budgetkey_data_pipelines/pipelines/procurement/tenders/exemptions/exemptions_scraper.py:
```diff
@@ -0,0 +1,137 @@
+                             "ctl00$m$g_cf609c81_a070_46f2_9543_e90c7ce5195b$ctl00$btnClear": "",
+                             "ctl00$m$g_cf609c81_a070_46f2_9543_e90c7ce5195b$ctl00$btnSearch": "\u05d7\u05e4\u05e9",
+                             "SearchFreeText": "\u05d7\u05e4\u05e9"})
+        # for some reason above code doesn't work
```
We can probably remove the code that's not working or being used, right?

akariv commented on tests/procurement/tenders/exemptions/fixtures/publisher_urls:
```diff
@@ -0,0 +1,48 @@
+pipelines/procurement/tenders/exemptions/add_publisher_urls_resource
+--
+{"mock": true}
```
Take a look at the work done here, perhaps it would be useful for mocking:
https://github.com/frictionlessdata/datapackage-pipelines/commit/748c067e87366b21404c39ff319c1240a964393f

OriHoch replied:
this is a different problem, I opened an issue to continue working on this: #9

OriHoch commented:
* removed unnecessary code + misc. refactoring
* added retry logic - by default retries 10 times with 1 minute between retries
* added max_pages parameter - allowing to limit the number of pages scraped per publisher
* didn't find an easy solution for the mocks - will be handled in #9

akariv commented on budgetkey_data_pipelines/pipelines/procurement/tenders/exemptions/exemptions_scraper.py:
```diff
@@ -0,0 +1,161 @@
+        for input_elt in self._page("#WebPartWPQ3 input"):
+            if input_elt.get("name"):
+                form_data[input_elt.attrib["name"]] = ""
+        # hard-code the form data with relevant elements from the html input elements
```
Everything before this is also redundant, no?

OriHoch replied:
it's not entirely unused, I'm collecting data from the input fields and then using it below

but, I agree that it's not clear, I'll refactor it a bit

OriHoch commented:
refactored a bit, added some comments
